docs: update readme to reflect new v1beta1 API changes (#612)
Azure/fleet: README.md

![GitHub go.mod Go version][5]
[![codecov][6]][7]

Fleet provides the following capabilities 

> * Join/Leave is a feature that allows a member cluster to join and leave a fleet(Hub cluster) in the fleet control plane.
> * Workload Orchestration where users can create resources on the hub cluster and then propagate these resources to all joined member clusters in the fleet.

Reviewer: ```suggestion
* Join/Leave is a feature that allows a member cluster to join and leave a fleet by registering a custom resource on the fleet's control plane (the hub cluster).
* Workload Orchestration where users can create resources on the hub cluster and then selectively propagate these resources to desired member clusters in the fleet.
```

> * Workload Orchestration where users can create resources on the hub cluster and then selectively propagate these resources to desired member clusters in the fleet.

Reviewer: ```suggestion
* Workload Orchestration is a feature that allows users to create resources on the hub cluster and then selectively propagate these resources to desired member clusters in the fleet.
```
